### DISCUSSION TITLE
How to set colorful theme for Office?

### DISCUSSION THREAD

**Discussion by Valthak** [AutoDarkMode/Windows-Auto-Night-Mode#525] [Q&A]
Dear All - as the title suggests, I wanted to ask how to set the colorful mode as the default light mode for the office package (change trigger is sunrise/sunset). I run windows eleven and Office 365. Below is the current configuration. You may notice that the Alternative light theme is unticked. It does not make a difference, if I leave the office setting on Use System Setting, it will use the White mode of office. I would prefer using the Colorful mode for office, which has colored headers and bars. Is this possible with the sunset/sunrise auto dark mode setting?

Thank you for you work, love the app.
Have a great day.

![image](https://user-images.githubusercontent.com/97289459/148538304-129b8cf3-e398-40ca-ac1e-dbf6466902c4.png)
![image](https://user-images.githubusercontent.com/97289459/148538489-f511fecd-28d6-485e-abd6-32923ae049ad.png)

**Comment by Spiritreader** [comment]
By default, "adapt to system" already uses byte value 0 (colorful) and 4 (black).

It looks like you are using the new Office anyway tho, so you can just set this to "disabled" and use the "use system setting" in offic.e
That should take care of it.

**Comment by Valthak** [comment]
Hi!
Thanks for the quick reply, I disabled in the app and set Office to 'Use System Setting', however it is still the white setting.
Thanks for the help!

**Comment by Talal-mah** [comment]
Hi I found the issue, it seems that the value in the log file is 5, where as the latest update to office brings the value of the "colourful theme to ID= 7. A simple edit of the config file should help you in this. 

You can manually check the theme values via reggedit : Computer\HKEY_CURRENT_USER\Software\Microsoft\Office\16.0\Common

@Spiritreader maybe include the new ID in the next update?

**Comment by Spiritreader** [comment]
the current values we use are

0 = colorful, 
3 = grey, 
4 = black, 
5 = white, 
6 = follow_system

The default configuration option are
- 0 (for light)
- 4 (for dark)

https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/blob/master/AutoDarkModeConfig/ComponentSettings/Base/OfficeSwitchSettings.cs#L6

The only instance where you should get a 5 in the log is if you enabled the "alternative light theme for office" checkbox here
![image](https://user-images.githubusercontent.com/18438899/151840639-f93698cf-8f0c-4248-84e6-f48813a612b1.png)

However, I do not believe it would make a lot of sense to change the default value. The Office Switch setting is only meant for legacy office versions that do not support the "follow system theme" option. And for these versions, 0 represents colorful.

Therefore my recommendation is to edit the configuration file if, for whatever reason it is not possible to use newer office's built in theme switch.

**Comment by boromyr** [comment]
I also have this problem with the version of Office with the new interface. I solved the problem by tracking registry changes with Registry Changes View during the change from "to color" to "black", which I saved as a .reg file and inserted into the script called by ADM.

**Comment by Talal-mah** [comment]
@Spiritreader Thanks for your comprehensive response. My issue with Office's automated "Use system settings" option is that the "light theme" is white and not colourful. Thus why I would like to use ADM for this change. 
I will try using the latest beta you have posted, and see what happens.

**Comment by Valthak** [comment]
@Spiritreader Hi - thanks for the explanation, I tried what you mentioned (and every other possible setting configuration), however it still does not work. I also changed the reg config to value 7, still nothing.

@Talal-mah can you please let me know how it worked in the beta version? Thank you.

**Comment by Talal-mah** [comment]
> @Talal-mah can you please let me know how it worked in the beta version? Thank you.

I can confirm that even using ADM Beta to change the office theme automatically, still, it goes to the white theme. Weird thing is that the selection doesn't even show up office. The drop-down is just blank. 
Screenshot for reference. 
![2022-03-01 09_48_51-Document1 - Word](https://user-images.githubusercontent.com/98762131/157322953-1dee5add-f532-4df9-986a-e0432e4874e8.png)

**Comment by Spiritreader** [comment]
The Office Switch is **ONLY** meant to be used with Office Versions up to 2019.
This is intentional.

As I have mentioned before:

> However, I do not believe it would make a lot of sense to change the default value. The Office Switch setting is only meant for legacy office versions that do not support the "follow system theme" option. And for these versions, 0 represents colorful.

It is only meant to be used with *old* *legacy* Office versions up to 2019
![image](https://user-images.githubusercontent.com/18438899/157329648-4b36252e-2042-4b5d-833c-10e4f88686a3.png)

If you have the "use system setting" option, then you **can not** use Auto Dark Mode anymore.

This is 100% intentional. If you truly desire to use ADM's office switch module for 365 2020 and onwards, you can still do so by modifying the config file and setting the right values as mentioned above.

There isn't really anything to fix, since it is legacy support for a reason.
I wonder if it would make more sense to remove the setting in Auto Dark Mode altogether to avoid confusion or attempt to detect the MS Office version and just disable it.

**Comment by Talal-mah** [comment]
> There isn't really anything to fix, since it is legacy support for a reason. I wonder if it would make more sense to remove the setting in Auto Dark Mode altogether to avoid confusion or attempt to detect the MS Office version and just disable it.

I think maybe detecting the version would be a great feature. 

The issue here is even if we try the mannual method you mention here; 
>This is 100% intentional. If you truly desire to use ADM's office switch module for 365 2020 and onwards, you can still do so by modifying the config file and setting the right values as mentioned above.
Still the correct colorful theme doesnt apply. I know its odd. 

I guess us O365 users are just out of luck...
Maybe we get Microsoft to change their default light theme to more Colorful than white when using the "use system settings" option.

ADM Config File for reference: 
![image](https://user-images.githubusercontent.com/98762131/157335347-200f6d8d-b2bd-43e4-8552-f9c2c5d4ff0c.png)

@Spiritreader

**Comment by Spiritreader** [comment]
> > There isn't really anything to fix, since it is legacy support for a reason. I wonder if it would make more sense to remove the setting in Auto Dark Mode altogether to avoid confusion or attempt to detect the MS Office version and just disable it.
> 
> I think maybe detecting the version would be a great feature.
> 
> The issue here is even if we try the mannual method you mention here;
> 
> > This is 100% intentional. If you truly desire to use ADM's office switch module for 365 2020 and onwards, you can still do so by modifying the config file and setting the right values as mentioned above.
> > Still the correct colorful theme doesnt apply. I know its odd.
> 
> I guess us O365 users are just out of luck... Maybe we get Microsoft to change their default light theme to more Colorful than white when using the "use system settings" option.
> 
> ADM Config File for reference: ![image](https://user-images.githubusercontent.com/98762131/157335347-200f6d8d-b2bd-43e4-8552-f9c2c5d4ff0c.png)
> 
> @Spiritreader

I am a bit confused now.
You said earlier that the new value for colorful is 7 for 2022 Office365.

But the screenshot you showed me has the LightTheme set to 0, not 7.

Also, can you get me a screenshot of the MS Office build number? It can be found in the Account tab, where you also change the theme, at the "About Word" section.
I'm running Verison 2108, which was updated on April 12th, and the default LightTheme when using the "use system setting" option is colorful.

**Comment by Talal-mah** [comment]
Hi @Spiritreader ,
Apologies for the confusion, What had happened was When I manually change the Light theme to 7 it worked a few versions ago, I haven't checked since then.
In the O365 version, it has always been the white theme for the light section in "use system settings" option in Office. 
Your version is slightly dated, as Microsoft releases updates almost every month for Office, some are significant, some are not. But usually there is a new Version number (not to be confused with build) atleast every two months. Try manually updating your office suite?

Please see my screenshot of my current version. It just updated a short while ago. 
![image](https://user-images.githubusercontent.com/98762131/167696597-b295e06e-bd03-41a4-946a-e2c4eaa161b7.png)
